feat: create ubuntu platform reference table (#211)
canonical/open-documentation-academy: ubuntu-desktop/reference/ubuntu-platform-reference.md.md

> # Ubuntu platform reference table

Reviewer: One thing to consider adding as a column is download size, or a typical range.

For example, the Ubuntu image for desktop is a lot larger than the image for WSL, which reflects the packages installed but also the constraints of the target platform and use case.

> The table below provides an overview of Ubuntu platforms, their features, use cases, and official links.

Author: Alright, the changes look good.

| --------------------- | -------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | **Ubuntu Desktop**    | Personal computers (laptops, desktops)                                     | <ul><li>GNOME-based GUI by default</li><li>Optimized for everyday computing</li><li>Extensive user-friendly ecosystem</li></ul>                                                                             | <ul><li>**Product page**: [ubuntu.com/desktop](https://ubuntu.com/desktop)</li></ul>                                                                                                                                              | <ul><li>General-purpose personal computing</li><li>Office productivity, browsing, media consumption</li></ul>                                             |

Reviewer: ```suggestion
| **Ubuntu Desktop**    | Personal computers (laptops, desktops)                                     | <ul><li>GNOME-based GUI by default</li><li>Optimized for everyday computing</li><li>Extensive range of software applications</li></ul>                                                                             | <ul><li>**Product page**: [ubuntu.com/desktop](https://ubuntu.com/desktop)</li></ul>                                                                                                                                              | <ul><li>General-purpose personal computing</li><li>Office productivity, browsing, media consumption</li></ul>                                             |
```

I think "extensive user-friendly ecosystem" was somewhat vague.

> | **Ubuntu Server**     | Physical or virtual servers, HPC clusters                                  | <ul><li>No GUI installed by default</li><li>Focus on stability and security</li><li>Long-term support releases available</li></ul>                                                                          | <ul><li>**Product page:** [ubuntu.com/server](https://ubuntu.com/server)</li><li>**Docs**: [https://documentation.ubuntu.com/server/](https://documentation.ubuntu.com/server/)</li></ul>                                         | <ul><li>Hosting web services, databases, or enterprise software</li><li>Infrastructure management</li></ul>                                               |

Reviewer: ```suggestion
| **Ubuntu Server**     | Physical or virtual servers, HPC clusters                                  | <ul><li>No GUI installed by default</li><li>Focus on stability and security</li><li>Long-term support releases available</li></ul>                                                                          | <ul><li>**Product page:** [ubuntu.com/server](https://ubuntu.com/server)</li><li>**Docs**: [https://documentation.ubuntu.com/server/](https://documentation.ubuntu.com/server/)</li></ul>                                         | <ul><li>Hosting web services, databases, or enterprise software</li><li>Infrastructure management</li></ul>                                               |
```

I think we need to be careful about suggesting there are features for certain platforms but not for others, when the truth is that they are available for both.

As one example, we suggest here that LTS releases are available for Server, but don't mention it for Desktop, Cloud, Core, WSL, etc. If there's something like this, which generally applies many cases, it might be worth mentioning at the top (before the table), something like, "Regardless of platform, Ubuntu offers Long term support releases [...]".

This is a general remark that you should keep in mind when revisiting this page.

| **Ubuntu on WSL**     | Machines running the Windows OS                                            | <ul><li>Run Ubuntu inside Windows</li><li>Command-line tools and utilities available natively</li><li>Seamless developer workflow</li></ul>                                                                 | <ul><li>**Product page**: [ubuntu.com/wsl](https://ubuntu.com/wsl)</li><li>**Docs:** [https://documentation.ubuntu.com/wsl/en/stable/](https://documentation.ubuntu.com/wsl/en/stable/)</li></ul>                                 | <ul><li>Development on Windows with Linux toolchains</li><li>Mixed environment scripting and testing</li></ul>                                            |

> | **Ubuntu Pro**        | Enterprise, production environments requiring compliance                   | <ul><li>Extended security maintenance (ESM)</li><li>10 years of security updates</li><li>FIPS 140-2 certified components</li></ul>                                                                          | <ul><li>**Product page**: [ubuntu.com/pro](https://ubuntu.com/pro)</li><li>**Docs:** [https://documentation.ubuntu.com/pro/](https://documentation.ubuntu.com/pro/)</li></ul>                                                     | <ul><li>Regulated industries: finance, healthcare</li><li>Long-term, secure deployments</li></ul>                                                         |

Reviewer: Ubuntu Pro is an interesting outlier here.

All the others are Ubuntu distributions but Pro is a feature that can be used with some Ubuntu distributions, to enhance them in specific ways.

I would suggest removing it from the table, but perhaps mentioning in a note after the table that Ubuntu supports Ubuntu Pro on several platforms, including x, y and z. You could then summarise the information about Pro that you included in the table.

Somewhat related, Ubuntu Frame is a display server rather than a distribution.

Alternatively, you could have an additional column that explains what the entry is, for example: variant of Ubuntu distribution with real-time kernel; a display server for embedded devices.

These could be common confusions for people, so it might be valuable to be maximally clear.

Author: So I took your suggestion and moved it from the table to the notes section. I think removing ubuntu frame would be more appropriate in this situation, less overhead that creating an entirely new column to justify the entry.

| **Real-time Ubuntu**  | Industrial, robotics, automotive, and other latency-sensitive environments | <ul><li>Real-time PREEMPT_RT kernel</li><li>Minimal latency, deterministic performance</li><li>Integration with Ubuntu security and updates</li></ul>                                                       | <ul><li>**Product page**: [https://ubuntu.com/real-time](https://ubuntu.com/real-time)</li><li>**Docs**: [https://documentation.ubuntu.com/real-time/en/latest/](https://documentation.ubuntu.com/real-time/en/latest/)</li></ul> | <ul><li>Robotics and industrial automation</li><li>Automotive control systems</li><li>High-performance audio production</li><li>Medical devices</li></ul> |

### Notes

> - **Flavors**: Beyond these main editions, there are official [Ubuntu flavors](https://ubuntu.com/desktop/flavours) (e.g., Kubuntu, Xubuntu, Ubuntu MATE, Ubuntu Budgie, Ubuntu Studio) each featuring a different desktop environment or specialized toolset.

Reviewer: ```suggestion
- **Flavors**: There are also official [Ubuntu flavors](https://ubuntu.com/desktop/flavours) (e.g., Kubuntu, Xubuntu, Ubuntu MATE, Ubuntu Budgie, Ubuntu Studio), each featuring a different desktop environment or specialized toolset.
```

Simplified.
